llvm/lib/Target/AArch64/AsmParser/AArch64AsmParser.cpp

```diff
@@ -7075,30 +7075,22 @@ bool AArch64AsmParser::parseDirectiveCPU(SMLoc L) {
 
     bool EnableFeature = !Name.consume_front_insensitive("no");
 
-    bool FoundExtension = false;
-    for (const auto &Extension : ExtensionMap) {
-      if (Extension.Name != Name)
-        continue;
+    auto It = llvm::find_if(ExtensionMap, [&Name](const auto &Extension) {
+      return Extension.Name == Name;
+    });
 
-      if (Extension.Features.none())
-        report_fatal_error("unsupported architectural extension: " + Name);
-
-      FeatureBitset Features = STI.getFeatureBits();
-      FeatureBitset ToggleFeatures =
-          EnableFeature
-              ? STI.SetFeatureBitsTransitively(~Features & Extension.Features)
-              : STI.ToggleFeature(Features & Extension.Features);
-      setAvailableFeatures(ComputeAvailableFeatures(ToggleFeatures));
-      FoundExtension = true;
-
-      break;
+    if (It != std::end(ExtensionMap)) {
+      if (EnableFeature)
+        STI.SetFeatureBitsTransitively(It->Features);
+      else
+        STI.ClearFeatureBitsTransitively(It->Features);
+    } else {
+      Error(CurLoc, "unsupported architectural extension: " + Name);
```

Comment:
Would it be simpler to just `return Error(..)`, such that the function would return `true` instead of `false` (by convention in the rest of this file and inline with `return Error`). I know it then only emits an error for the first extension that is unsupported, rather than do so for all of them, but I'm not sure if this is important and it simplifies the logic a bit.

I spotted that the case on line 7064 currently returns `false`, which I think should be `true` (or simply `return Error(..)` as well.

Reply:
I don't mind. I originally tried to keep it nfc in that regard, happy to change it as well as the tests but I have no strong opinion either way. Lemme know

Reply:
I would prefer returning directly because it's structurally simpler and helps avoid any future issues with the code that follows after the `Error`. For example, the function returns `false` (no error), where it should have returned `true`. This isn't much of a problem at the moment, because nothing checks the return value of this function, but if it would this would not be correct.